cmd/sonobuoy/app/gen_plugin.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/heptio/sonobuoy/pkg/config"
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/heptio/sonobuoy/pkg/plugin"
+	"github.com/heptio/sonobuoy/pkg/plugin/loader"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	placeholderHostname = "<hostname>"
+)
+
+// GenPluginConfig are the input options for running
+type GenPluginConfig struct {
+	Paths      []string
+	PluginName string
+}
+
+var genPluginOpts GenPluginConfig
+
+func init() {
+	cmd := &cobra.Command{
+		Use:    "plugin",
```

Comment:
This doesn't look like a sub-command of gen, but it's own sub-command of the root?  Is that what you want?

Reply:
I add it to GenCommand, so I think it'll be a subcommand of that, no?